src/main/java/com/mystays/authorizationserver/controller/AuthenticationController.java

```diff
@@ -0,0 +1,60 @@
+package com.mystays.authorizationserver.controller;
+
+import com.mystays.authorizationserver.config.CustomAuthenticationManager;
+import com.mystays.authorizationserver.entity.User;
+import com.mystays.authorizationserver.model.AuthenticationRequest;
+import com.mystays.authorizationserver.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@RestController
+public class AuthenticationController {
+
+    @Autowired
```

Comment:
you can have a single constructor injection instead of separate field injections.